### PR TITLE
Add support for Connector/J 6.x

### LINKED ISSUE #18: Bump mysql-connector to 6.x
Can you bump the sql-connector to the latest version? We were happy with your library because it allowed us to get away from this approach here (http://labs.unacast.com/2016/02/18/cloud-sql-on-google-container-engine/), but as we are now bumping our dependencies it fails and we seem forced to revert back, increasing build time and container complexity / responsibilities. 

The error is ```java.lang.NoClassDefFoundError: com/mysql/jdbc/SocketFactory```. I believe it is pretty simple - the functionality is still present in 6.X, it just put in another package: ```com.mysql.cj.api.io.SocketFactory```

On a side note: Is there any rationale why the ```com.google.cloud.sql.mysql.SslSocketFactory``` is package local? It would be a lot easier for me to just rewrite this by creating my own SocketFactory without having to copy and duplicate your SSL and authentication stuff.

### PR DESCRIPTION
**Pull request by laixer**
Connector/J 6.x has a different location for the SocketFactory interface. 
This was done by splitting the project into multiple modules:
  core - mysql driver agnostic library code
  connector-j-5 - module for the 5.x version of Connector/J
  connector-j-6 - module for the 6.x version of Connector/J

The connector-j-6 module publishes a new artifact with id
'mysql-socket-factory-connector-j-6'.

Fixes #18.

### CODE REVIEW

File: connector-j-6/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.mysql;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.net.SocketException;
```
Comment (hfwang) [inline connector-j-6/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java]: nit: unused import

**Comment by hfwang** [comment]
er, lack of proofreading: "it looks like keeping the split in the artifact id makes more sense for us anyway"

File: connector-j-6/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    String instanceName = props.getProperty("cloudSqlInstance");
+    checkArgument(
+            instanceName != null,
+            "cloudSqlInstance property not set. Please specify this property in the JDBC URL or "
+                    + "the connection Properties with value in form \"project:region:instance\"");
+
+    logger.info(String.format("Connecting to Cloud SQL instance [%s].", instanceName));
+
+    this.socket = SslSocketFactory.getInstance().create(instanceName);
+    return socket;
```
Comment (laixer) [inline connector-j-6/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java]: Done.

File: connector-j-6/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java (hunk 1 above)
Comment (laixer) [inline connector-j-6/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java]: Done.

**Comment by hfwang** [comment]
LGTM

**Pull request merged**